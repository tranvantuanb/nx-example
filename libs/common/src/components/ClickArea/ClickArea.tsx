import React from 'react';
import tw, { styled } from 'twin.macro';

export type ClickAreaProps = React.HTMLAttributes<HTMLDivElement>;

const StyledClickArea = styled.div`
  ${tw`absolute w-full h-full top-0 left-0`}
`;

export const ClickArea: React.FC<ClickAreaProps> = React.memo((props) => {
  return <StyledClickArea {...props} />;
});
