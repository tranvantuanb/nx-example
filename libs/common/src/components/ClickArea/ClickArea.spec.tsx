import { render } from '../../test';

import { ClickArea, ClickAreaProps } from './ClickArea';

describe('ClickArea', () => {
  it('should render successfully', () => {
    const defaultProps: ClickAreaProps = {};
    const { container } = render(<ClickArea {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
