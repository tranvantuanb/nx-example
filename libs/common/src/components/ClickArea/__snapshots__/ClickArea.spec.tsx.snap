// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickArea should render successfully 1`] = `
<div>
  <div
    class="css-4k0i9f-StyledClickArea ectarx80"
  />
</div>
`;
